resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -1470,6 +1470,11 @@ private[spark] object Client extends Logging {
       addClasspathEntry(getClusterPath(sparkConf, cp), env)
     }
 
+    val cpSet = extraClassPath match {
+      case Some(classPath) => classPath.split(File.pathSeparator).toSet
```

Comment:
Actually this is risky because it loses the order of classpaths. This matters when multiple jars are provided somehow in the production env. Should use sth like LinkedHashSet (see also https://github.com/HyukjinKwon/spark/commit/c6b8eb71a9638c9a8ce02d11d5fe26f4c5be531e)

Reply:
Oh, nvm. okay, it is just used for checking if sth exists. All good.

Reply:
Do you mean `extraClassPath.getOrElse("").split(File.pathSeparator).toSet`?

How about 

```
    val cpSet = extraClassPath match {
      case Some(classPath) if Utils.isTesting =>
        classPath.split(File.pathSeparator).toSet
      case _ => Set.empty[String]
    }
```

Reply:
If only for testing, we can split only when `Utils.isTesting` is true